docs/key-issues/gpus.md

```diff
@@ -180,3 +204,87 @@ Potential solutions on the PyPI side include:
 - make an environment marker or selector package approach work,
 - improve interoperability with other package managers, in order to be able to
   declare a dependency on a CUDA or ROCm version as externally provided,
+
+
+## Additional Notes on CUDA Compatibility
+
+Compatibility across CUDA versions is a common problem with numerous pitfalls to be aware of.
+There are three primary components of CUDA:
+
+1. The CUDA Toolkit (CTK): This component includes the CUDA runtime library
+   (libcudart.so) along with a range of other libraries and tools including
+   math libraries like cuBLAS. libcudart.so and a few other core headers and
+   libraries are the bare minimum required to compile and link CUDA code.
+2. The User-Mode Driver (UMD): This is the libcuda.so library. This library is
+   required to load and run CUDA code.
+3. The Kernel-Mode Driver (KMD): The nvidia.ko file. This constitutes what is
+   typically considered a "driver" in common parlance when referring to other
+   peripherals connected to a computer.
+
+CUDA developers and users usually do not need to be aware of the KMD.
+For the rest of this section, when referring to the "driver" we will always be referring to the UMD.
```

Comment:
Sorry, I still don't feel comfortable about the claim that "don't need to be aware of KMD." People who own a workstation under their desk (like I did) would likely disagree, because this bites users from time to time. Can we rephrase it this way? 
```suggestion
As far as MVC is concerned, KMD can be ignored providing that it meets [CUDA's requirement](https://docs.nvidia.com/deploy/cuda-compatibility/index.html).
For the rest of this section, therefore, the "driver" and the UMD are considered interchangeable.
```

Reply:
Fair enough, you're right. I've definitely observed >0 issues with that before, so it can't be ignored altogether. I'll adopt your rewriting of the first sentence.